app/jobs/award_badge_job.rb

```diff
@@ -16,4 +16,10 @@ def perform(user, badge_slug, send_email: true)
   def self.badge(badge_slug)
     "Badges::#{badge_slug.to_s.camelize}Badge".constantize
   end
+
+  def self.worth_queuing_context(context)
+    return {} if context.nil?
+
+    { context.class.base_class.name.demodulize.underscore => context }.symbolize_keys
```

Comment:
I need the `base_class` name otherwise I would never get `reputation_token` as the keyword, as it would be `code_merge_token` for one case, but `exercise_author_token` for another. 

The `demodulize` is used to remove any namespace (e.g. `User::ReputationToken` becomes `ReputationToken`).